pkg/oc/admin/diagnostics/pod.go

```diff
@@ -89,37 +80,26 @@ func (o *PodDiagnosticsOptions) Complete(args []string) error {
 	return nil
 }
 
-// BuildAndRunDiagnostics builds diagnostics based on the options and executes them, returning a summary.
-func (o PodDiagnosticsOptions) BuildAndRunDiagnostics() (bool, error, int, int) {
-	failed := false
-	errors := []error{}
-	diagnostics := []types.Diagnostic{}
+// BuildAndRunDiagnostics builds diagnostics based on the options and executes them, returning fatal error(s) only.
+func (o PodDiagnosticsOptions) BuildAndRunDiagnostics() error {
+	var fatal error
+	var diagnostics []types.Diagnostic
 
 	func() { // don't trust discovery/build of diagnostics; wrap panic nicely in case of developer error
```

Comment:
this looks a bit unusual - is this in case of a runtime error?

Reply:
Yes. My thinking was that when a user runs a diagnostic, they're probably already facing a frustrating problem, and the most frustrating thing in the world would be to have the diagnostic tool itself completely bomb out. And given the system being diagnosed is probably broken, the likelihood of unexpected conditions leading to panics is much higher than during "normal" operation where most diagnostic development occurs.

Since individual diagnostics do orthogonal things, it seemed acceptable to recover from panics and at least give the user *something* beyond just an infuriatingly obtuse stack trace.

This has been this way since the beginning of diagnostics, and deads wasn't super fond of it then either, but it still seems right to me.

Reply:
Maybe you could set this up not here, but rather inside the `commandRunFunc` this way each command would not have to do it? Or generally at a higher level. 👍 for the reasons above.

Reply:
To me, that doesn't quite match the conceptual level of where the problem is being isolated. Diagnostics come in several "areas" (avoiding overloading the terms "class", "kind", "type") -- client, cluster, host, etc. -- and the build routine for each has its own panic recovery; so one failing would still allow other areas to build and run. Additionally there is a panic recovery around the run of each individual diagnostic -- again, to maximize what can run in a diagnostic situation. Kicking it up to the command level would leave little improvement over just letting the panic halt execution entirely.

I can think of some further refactoring to get rid of the "areas" and put the panic isolation at clearer points, but I don't want this PR to wait on that.